EIPS/eip-4444.md

```diff
@@ -0,0 +1,101 @@
+---
+eip: 4444
+title: Bound Historical Data in Execution Clients
+description: Prune historical data in clients older than one year
+author: George Kadianakis (@asn-d6), lightclient (@lightclient), Alex Stokes (@ralexstokes)
+discussions-to: https://ethereum-magicians.org/t/eip-4444-bound-historical-data-in-execution-clients/7450
+status: Draft
+type: Standards Track
+category: Networking
+created: 2021-11-02
+---
+
+## Abstract
+
+Clients must stop serving historical blocks/receipts that are older than a year on the p2p layer. Clients may prune such historical data.
+
+## Motivation
+
+Historical blocks, states, and receipts currently occupy more than 400GB of disk space (and growing!). Therefore, to validate the chain, users must typically have a 1TB disk.
+
+Historical data is not necessary for validating new blocks, so once a client has synced the tip of the chain, historical data is only retrieved when requested explicitly over the JSON-RPC or when a peer attempts to sync the chain. By pruning the history, this proposal reduces the disk requirements for users. Pruning history also allows clients to remove code that processes historical blocks. This means that execution clients don't need to maintain code paths that deal with each upgrade's compound changes.
+
+This proposal also concretizes the availability guarantees of the p2p layer as clients won't be able to fetch historical data anymore. This results in less strain on the network as clients adopt more lightweight sync strategies based on the PoS weak subjectivity assumption.
+
+## Specification
+
+| Parameter | Value | Description |
+| - | - | - |
+| `HISTORY_PRUNE_EPOCHS` | 82125 | A year in beacon chain epochs |
+
+Clients MUST NOT serve blocks/states/receipts that are older than `HISTORY_PRUNE_EPOCHS` epochs on the p2p network.
+
+Clients MAY locally prune block/state/receipt history that is older than `HISTORY_PRUNE_EPOCHS` epochs.
+
+#### Bootstrapping and syncing
+
+This EIP impacts the way clients bootstrap and sync. Clients won't be able to full sync or snapshot sync since historical data won't be available on the p2p network.
+
+Clients MUST use a valid Weak Subjectivity Checkpoint to bootstrap from a more recent view of the chain. For the purpose of syncing, clients treat weak subjectivity checkpoints as the genesis block. We call this method "checkpoint sync".
+
+For the purposes of this proposal, we assume clients always start with a configured and valid weak subjectivity checkpoint. The way this is achieved is out-of-scope for this proposal.
```

Comment:
Actually back in April 2020 or so, within the Ipsilon team we thought about making a proposal to just hardcode given hashes for given blocks in an EIP. The idea would be to hardcode the hashes for past hard forks.

However then regenesis was proposed, which in practice does the same, but programatically.

Since regenesis as a concept is delayed ever so often, and given this EIP, would such a proposal make sense now?

Reply:
Moving to magicians thread.